src/instructlab/model/list.py

```diff
@@ -0,0 +1,111 @@
+from requests import exceptions as requests_exceptions
+import click
+import pandas
+
+# First Party
+from instructlab import clickext
+from instructlab.configuration import DEFAULTS
+from instructlab.model.backends.backends import is_model_gguf
+
+logger = logging.getLogger(__name__)
+
+
+@click.command(name="list")
+@clickext.display_params
+@click.option(
+    "--model-dirs",
+    help="Base directories where models are stored.",
+    multiple=True,
+    default=[DEFAULTS.MODELS_DIR],
+    show_default=True,
+)
+@click.option(
+    "--list-checkpoints",
+    type=click.BOOL,
+    default=False,
+    is_flag=True,
+    show_default=True,
+)
+def model_list(model_dirs: list[str], list_checkpoints: bool):
+    """lists models"""
+    t = PrettyTable(["Model Name", "Model Age", "Model Size"], hrules=ALL)
+    for dir in model_dirs:
+        for entry in Path(dir).iterdir():
+            # if file, just tally the size. This must be a GGUF.
+            if entry.is_file() and is_model_gguf(entry):
+                # stat the gguf, add it to the table
+                stat = Path(entry.absolute()).stat(follow_symlinks=False)
+                f_size = stat.st_size
+                f_size, magnitude = convert_bytes_to_proper_mag(f_size)
+                # add to table
+                age = pandas.to_timedelta(time.time_ns() - stat.st_ctime_ns)
+                t.add_row([entry.name, age, f"{f_size} {magnitude}"])
+            elif entry.is_dir():
+                actual_model_name = ""
+                all_files_sizes = 0
+                add_model = False
+                # walk entire dir.
+                for root, _, files in os.walk(entry.as_posix()):
+                    normalized_path = os.path.normpath(root)
+                    # Split the path into its components
+                    parts = normalized_path.split(os.sep)
+                    # Get the last two parts and join them back into a path
+                    last_two_parts = os.path.join(parts[-2], parts[-1])
+                    # if this is a dir it could be:
+                    # top level repo dir `instructlab/`
+                    # top level model dir `instructlab/granite-7b-lab`
+                    # checkpoint top level dir `step-19`
+                    # any lower level dir: `instructlab/granite-7b-lab/.huggingface/download.....`
+                    # so, query HF and see if its a model. For NOW only list models with HF's `ModelCard`.
+                    # if --list-checkpoints is specified, we will list all checkpoints in the checkpoints dir regardless of the ModelCard
+                    try:
+                        ModelCard.load(last_two_parts)
+                        actual_model_name = last_two_parts
+                        all_files_sizes = 0
+                        add_model = True
+                    except (ValueError, requests_exceptions.HTTPError):
+                        if list_checkpoints and dir is DEFAULTS.CHECKPOINTS_DIR:
```

Comment:
You don't have to change this but here's a tip I like to share since it leads to cleaner branching logic:
```py
# invert the condition
if not list_checkpoints or dir is not DEFAULTS.CHECKPOINTS_DIR:
    continue

# then you can continue here in the main
logging.debug("Including model regardless of ModelCard validity")
```